org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/Launcher.java

```diff
@@ -294,7 +299,7 @@ public Builder<T> validateMessages(boolean validate) {
 		
 		public Builder<T> traceMessages(PrintWriter tracer) {
 			if (tracer != null) {
-				this.messageTracer = new MessageTracer(tracer);
```

Comment:
With your current approach, we'd still need to keep this line with the one-arg constructor as a fallback. Otherwise, an existing subclass that completely overrides `Builder.create()` would have no `messageTracer` set, which is a breaking change. Similarly, we'd need to make the new two-arg constructor public so that such subclasses could reset the default `messageTracer` in their `create()` implementations. That's why I much prefer your previous approach with `MessageTracer.setJsonHandler`. It is so much less of a hassle.